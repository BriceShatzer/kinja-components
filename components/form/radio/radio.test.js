import * as React from 'react';
import { shallow, mount } from 'enzyme';

import Radio from './radio';


describe('<Radio />', () => {
	const onClick = jest.fn();

	const stubElement = ({
		checked = false,
		label = 'label',
		name = 'name',
		value = 'value'
	} = {}) => (
		<Radio
			checked={checked}
			label={label}
			name={name}
			onClick={onClick}
			value={value}
		/>
	);

	it('should render properly', () => {
		const wrapper = shallow(stubElement());
		expect(wrapper).toMatchSnapshot();
	});

	it('should trigger an onChange when it is checked', () => {
		const wrapper = mount(stubElement());
		wrapper.find('input').simulate('click');
		expect(onClick).toHaveBeenCalledTimes(1);
		expect(onClick).toHaveBeenCalledWith('value');
	});
});