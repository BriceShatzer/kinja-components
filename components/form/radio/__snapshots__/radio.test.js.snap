// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> should render properly 1`] = `
<EnsureDefaultTheme>
  <radio__Wrapper
    htmlFor="value"
    onClick={[Function]}
  >
    <radio__Input
      checked={false}
      id="value"
      name="name"
      readOnly={true}
      value="value"
    />
    <radio__Checkmark />
    <radio__Label>
      label
    </radio__Label>
  </radio__Wrapper>
</EnsureDefaultTheme>
`;
